packages/js/src/ui/components/Notification/DefaultNotification.tsx

```diff
@@ -14,20 +16,49 @@ const NotificationBody = (props: NotificationBodyProps) => {
 };
 
 type DefaultNotificationProps = {
-  notification: InboxNotification;
+  notification: Notification;
+  onNotificationClick?: NotificationClickHandler;
+  onActionClick?: NotificationActionClickHandler;
 };
 
 //TODO: Complete the implementation
 export const DefaultNotification = (props: DefaultNotificationProps) => {
   const style = useStyle();
   const { t, locale } = useLocalization();
 
+  const handleNotificationClick: JSX.EventHandlerUnion<HTMLAnchorElement, MouseEvent> = (e) => {
+    e.stopPropagation();
+    e.preventDefault();
+
+    if (!props.notification.isRead) {
+      props.notification.read();
+    }
+
+    props.onNotificationClick?.({ notification: props.notification });
+    if (props.notification.redirect?.url) {
+      window.open(props.notification.redirect?.url, '_blank', 'noreferrer noopener');
+    }
+  };
+
+  const handleActionButtonClick = (action: ActionTypeEnum, e: MouseEvent) => {
+    e.stopPropagation();
+
+    if (action === ActionTypeEnum.PRIMARY) {
+      props.notification.completePrimary();
+    } else {
+      props.notification.completeSecondary();
+    }
+
+    props.onActionClick?.({ notification: props.notification, action });
+  };
+
   return (
-    <div
+    <a
       class={style(
         'notification',
-        'nt-w-full nt-text-sm hover:nt-bg-neutral-100 nt-group nt-relative nt-flex nt-px-6 nt-py-4 nt-gap-2'
+        'nt-w-full nt-text-sm hover:nt-bg-neutral-100 nt-group nt-relative nt-flex nt-px-6 nt-py-4 nt-gap-2 nt-cursor-pointer'
```

Comment:
❓  Do we want a pointer at all times here since it will mark it as read anyway?

Reply:
No, we don't, only when the notification is unread or there is a redirect URL.